### PR TITLE
[fixes #19392881] Validate empty/zero values for funding flows

### PR DESCRIPTION
**Pull request by technogoat**

### CODE REVIEW

File: spec/models/funding_flow_spec.rb [hunk 1]
```diff
@@ -165,18 +165,26 @@ describe FundingFlow do
       basic_setup_project
     end
 
-    it "should validate the spend fields" do
```
Comment (brvtl) [inline spec/models/funding_flow_spec.rb:179]: You're not checking the case that the budget is present and the spend is nil or the case that the spend is present and the budget is nil.

File: spec/models/funding_flow_spec.rb [hunk 2]
```diff
@@ -165,18 +165,43 @@ describe FundingFlow do
...
+    it "should validate one or the other" do
       @funding_flow = Factory.build(:funding_flow,
-                                    :project => @project, :budget => 'abcd',
+                                    :spend => nil,
+                                    :budget => 1,
+                                    :project => @project,
                                     :from => @organization)
-      @funding_flow.save.should be_false
+
+      @funding_flow_2 = Factory.build(:funding_flow,
```
Comment (dalibor) [inline spec/models/funding_flow_spec.rb]: use the same object @funding_flow

File: spec/models/funding_flow_spec.rb [hunk 3]
```diff
@@ -165,18 +165,43 @@ describe FundingFlow do
...
+                                    :spend => 1,
+                                    :budget => nil,
+                                    :project => @project,
+                                    :from => @organization)
+
+      @funding_flow.save.should == true
```
Comment (dalibor) [inline spec/models/funding_flow_spec.rb]: use valid?, no need to save the object to db

File: app/models/funding_flow.rb [hunk 4]
```diff
@@ -24,8 +24,10 @@ class FundingFlow < ActiveRecord::Base
   # also see validations in BudgetSpendHelper
   #validates_presence_of :project_id # See: https://rails.lighthouseapp.com/projects/8994-ruby-on-rails/tickets/2815-nested-models-build-should-directly-assign-the-parent
                                      # and workaround is in project.rb
-  validates_numericality_of :spend, :if => Proc.new {|ff| ff.spend.present?}
-  validates_numericality_of :budget, :if => Proc.new {|ff| ff.budget.present?}
+  validates_numericality_of :spend, :greater_than => 0,
+    :if => Proc.new { |is| is.project.nil? && is.spend.present? && (!is.budget.present? || is.budget == 0)  }
```
Comment (dalibor) [inline app/models/funding_flow.rb]: why do we need is.project.nil? check?

**Comment by brvtl** [comment]
merged

**Pull request closed without merging**